### DISCUSSION TITLE
RPi 4 + Coral?

### DISCUSSION THREAD

**Discussion by mario-tux** [blakeblackshear/frigate#1295] [General]
Hi, I'm using Frigate on an Intel  i3-4150@3.5GHz + USB Coral. Using Quicksync I can manage 5 cameras with a resolution of 2592x1944 at 5 fps in H264 encoding with a quiet low CPU usage (about 100% of 1 core).

I would like to move Frigate on a dedicated device with a low power profile: I'm was thinking about a RPi 4 with at least 2 GB of RAM running with 64 bit code. The documentation says "10-15ms | Can handle a small number of cameras. The 2GB version runs fine". Can it manage my 5 cameras? Power usage?

My cameras are able to use H265 for encoding (saving bandwith) but my i3 generation is not able to support the hardware decoding. Is able RPi 4 manage H265 hardware decoding? Is it really bad the software one? What low power/low cost device can manage H265 hardware decoding?

Thanks.

**Comment by blakeblackshear** [comment]
The RPi4 won't be able to manage decoding that many streams at that resolution with hwaccel because it has limited amounts of GPU ram. It might be able to do it using the CPU, but I can't say for sure.

H265 may save bandwidth, but it will take more CPU power to decode and the clips/recordings won't playback in a browser.

I would look at a low power device with an Intel processor that supports quicksync. I can't say for sure which devices may be powerful enough for 5 cameras at that resolution.

**Comment by b3nj1** [comment]
FWIW, I'm using an RPi 4 + Coral with:
- no hwaccel options (tried it once briefly and it didn't work, so I bailed)
- 11 cameras doing detect at 5 fps at VGA (or near depending on the camera) using H264 
  - the cameras are set for their sub streams to the low res and low frame rate so that the Pi doesn't have to do the work to scale or drop frames; without this, the Pi is over burdened
- 3 of those cameras are set for clips and 24/7 record at 15fps UHD/3840x1920 res using H264
- 1 of those cameras is set for clips and 24/7 record at 15fps 2560x1920 res using H264
- the remaining 7 cameras are set for clips using the low res/rate detect stream, and are not set for recording
  - It might be possible to set the clips to use these camera's full res/rate stream. I'll try at some point. 

The system load is generally in the 3.5-4.5 range, and the detection results and latency seem fine.

If you don't have tiny objects to detect, you could do detect on low res streams to really help on loading.

- Benjamin

**Reply by blakeblackshear** [reply]
You can probably enable hwaccel for a few cameras to further decrease load. It won't be able to do it for all of them, but use it where you can.

**Reply by alanpilz** [reply]
@b3nj1 Good info.. I am trying to get 7 cameras connected to my RPi4 4Gb Ram + Coral.. My cameras are much lower res than yours main stream 1920x1080 and sub stream 640x480. I have enabled hwaccel for all the cameras and it works for 2 cameras but adding anymore seems to overload my RPi ... I have just got a google coral and tried to use that.. BUT with the Coral my HA becomes non responsive after a few minutes of starting Frigate with the same 2 cameras that work without TPU configured. The docs don't mention anything about Coral installation, Frigate logs suggest TPU was detected... but am I missing something? Did you simply just plug in the coral or did you install anything beforehand? The Getting Started for Coral shows a script to install a Debian package, but not sure if this applies for Home Assistant OS? What did you do ? (I will try disabling hwaccel for all of the cameras).

Currently back to running 2 cameras with hwaccel without Coral to make sure the HA issue is not relating to anything else that's going on.

_Alan_

**Reply by alanpilz** [reply]
Disabling hwaccel seems to have resolved my issues... 7 cameras added and everything appears to be ok

**Comment by b3nj1** [comment]
Glad it worked with disabling hwaccel. The TPU just worked, but sometimes I
have to restart docker because it isn’t detected.  I put a little watchdog
script to check the log for the error and restart automatically.

For the record, I was able to get hwaccel to function. I had to turn
privileged to true for docker. I enabled it initially on my 4 cameras that
are recording 24/7. But, it *seemed* to actually increase the load. I
turned it off, and will collect load data for a few days. Then, I’ll try it
on for a few days.

**Comment by alanpilz** [comment]
> Glad it worked with disabling hwaccel. The TPU just worked, but sometimes I
> have to restart docker because it isn’t detected.  I put a little watchdog
> script to check the log for the error and restart automatically.
> 
> For the record, I was able to get hwaccel to function. I had to turn
> privileged to true for docker. I enabled it initially on my 4 cameras that
> are recording 24/7. But, it *seemed* to actually increase the load. I
> turned it off, and will collect load data for a few days. Then, I’ll try it
> on for a few days.
> 

Well I think I spoke too soon... left it running over night and this morning I'm not able to access home assistant from the app. Loading loading loading.... after waiting long enough, I eventually logged in with a message saying the Frigate Add On isn't running. So it must have somehow terminated. Started it again, and now loading loading loading again.. where can I find logs to troubleshoot... (sorry newbie question) but historic frigate logs are not available in the GUI?

**Comment by b3nj1** [comment]
I’m not running it via the add-on. I have a dedicated Pi 4 running it in
docker. So, I’m not sure how to get those logs.
